### PR TITLE
[DependencyInjection] provide better error message when using deprecated configuration options

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14687 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php [hunk 1]
```diff
@@ -201,14 +202,17 @@ private function parseDefinition($id, $service, $file)
         }
 
         if (isset($service['factory_class'])) {
+            trigger_error(sprintf('The "factory_class" option in file "%s" is deprecated since version 2.6 and will be removed in 3.0. Use Definition::setFactory() instead.', $file), E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: As we now it comes from a YAML file, the hint is not the right one. We should tell the dev how to configure it in a YAML context.

Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php]: Oh indeed, bad copy and paste. Fixed that.

**Comment by Gladhon** [comment]
+1

**Comment by xabbuh** [comment]
ping @symfony/deciders

File: src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php [hunk 2]
```diff
@@ -137,6 +137,9 @@ public function testLegacyLoadServices()
         $this->assertEquals('getInstance', $services['factory_service']->getFactoryMethod());
     }
 
+    /**
+     * @group legacy
+     */
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: Why have you tagged this one as legacy?

Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: I mean, I understand why, but as all assertions are not about deprecated features, you should probably extract the tests for deprecated features in another test.

Comment (stof) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: indeed, this looks weird to me

Comment (xabbuh) [inline src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php]: You're right, moving this particular service to the legacy file makes indeed more sense.

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**